src/clr-angular/data/datagrid/datagrid-column-toggle.spec.ts

```diff
@@ -271,6 +275,30 @@ export default function(): void {
                 expect(columnCheckboxes[columnCheckboxes.length - 1].disabled).toBe(false);
             });
         });
+
+        describe("Custom buttons", function() {
+            // Until we can properly type "this"
+            let context: TestContext<ClrDatagridColumnToggle, CustomButtonsTest>;
+
+            beforeEach(function() {
+                context = this.create(ClrDatagridColumnToggle, CustomButtonsTest,
+                                      [HideableColumnService, ColumnToggleButtonsService]);
+            });
+
+            it("projects custom buttons and title", function() {
+                const iconBtn = context.clarityElement.querySelector(".column-switch-wrapper > button");
+                expect(context.clarityDirective.open).toBe(false);
+                iconBtn.click();
+                context.detectChanges();
+                expect(context.clarityDirective.open).toBe(true);
+
+                const buttons = context.clarityElement.querySelectorAll(".switch-footer button");
+                expect(buttons[0].innerText).toEqual("Select All!".toUpperCase());
+                expect(buttons[1].innerText).toEqual("OK!".toUpperCase());
```

Comment:
Do we want a unit test to validate the defaults are coming through?

Reply:
Added a test.